ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/UserManagementBasedAuthorizer.scala

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.auth
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+import com.daml.caching.{CaffeineCache, ConcurrentCache}
+import com.daml.error.definitions.LedgerApiErrors
+import com.daml.error.{ContextualizedErrorLogger, DamlContextualizedErrorLogger}
+import com.daml.ledger.api.domain.UserRight
+import com.daml.ledger.participant.state.index.v2.UserManagementStore
+import com.daml.ledger.participant.state.index.v2.UserManagementStore.{Result, UserNotFound}
+import com.daml.lf.data.Ref
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.github.benmanes.caffeine.cache.Caffeine
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+/** Validates authorizaton claims against the set of known users and their users.
+  */
+trait UserManagementBasedAuthorizer {
+  def validate(claims: ClaimSet.Claims): Either[AuthorizationError, Unit]
+}
+
+// TODO participant user management: Review usages
+object NoOpUserManagementBasedAuthorizer extends UserManagementBasedAuthorizer {
+  override def validate(claims: ClaimSet.Claims): Either[AuthorizationError, Unit] = Right(
+    ()
+  )
+}
+
+// TODO participant user management: Write tests
+class CachedUserManagementBasedAuthorizer(
+    delegate: UserManagementBasedAuthorizer,
+    expiryAfterWriteInSeconds: Int,
+) extends UserManagementBasedAuthorizer {
+
+  private val usersCache: ConcurrentCache[ClaimSet.Claims, Either[AuthorizationError, Unit]] =
+    CaffeineCache[ClaimSet.Claims, Either[AuthorizationError, Unit]](
```

Comment:
As I understand this cache is for claims resolution and allows you not to process claims unless time elapsed

Reply:
I couldn't see what would be the benefit from wrapping an instance of `UserManagementStore` in another layer cache since that instance should already be a `CachedUserManagementStore` (which caches db accesses). My understanding is that, if we nevertheless were to do that, the outer cache would be just caching the inner cache hits which seems pointless (I'm assuming the same cache entry retention times for both caches).

Rather, I considered what is the question&answer that we are directly interested in for the user based authorizer, which I determined to be: given some claims (`ClaimSet.Claims`) is the user management still authorizing them (`Either[AuthorizationError, Unit]`) - hence `ConcurrentCache[ClaimSet.Claims, Either[AuthorizationError, Unit]]`.
The benefit here is in saving some CPU cycles on calculating `Either[AuthorizationError, Unit]`.

Reply:
Will remove cache and change to throttling